sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -3356,21 +3356,11 @@ class Dataset[T] private[sql](
    * user-registered callback functions.
    */
   private def withAction[U](name: String, qe: QueryExecution)(action: SparkPlan => U) = {
-    try {
-      qe.executedPlan.foreach { plan =>
-        plan.resetMetrics()
-      }
-      val start = System.nanoTime()
-      val result = SQLExecution.withNewExecutionId(sparkSession, qe) {
-        action(qe.executedPlan)
-      }
-      val end = System.nanoTime()
-      sparkSession.listenerManager.onSuccess(name, qe, end - start)
-      result
-    } catch {
-      case e: Exception =>
-        sparkSession.listenerManager.onFailure(name, qe, e)
-        throw e
+    qe.executedPlan.foreach { plan =>
```

Comment:
can this throw an exception? Imagine if `df.count()` threw an exception, and then you run it again.
Won't this be a behavior change in that case?

Reply:
I don't think `resetMetrics` can throw exception...

Reply:
can't executedPlan throw an exception? I thought it can if the original spark plan failed?

Reply:
ah i see your point here